response_operations_ui/common/respondent_utils.py

```diff
@@ -0,0 +1,34 @@
+def status_enum_to_string(status):
+    dict = {
+        'ACTIVE': 'Active',
+        'CREATED': 'Created',
+        'SUSPENDED': 'Suspended',
+        'LOCKED': 'Locked'
+    }
+    return dict.get(status, None)
+
+
+def status_enum_to_class(status):
+    dict = {
+        'ACTIVE': 'status--success',
+        'CREATED': 'status--info',
+        'SUSPENDED': 'status--warning',
+        'LOCKED': 'status--dead'
+    }
+    return dict.get(status, None)
+
+
+def filter_respondents(respondents):
+    filtered_respondents = []
+    for respondent in respondents:
+        try:
+            filtered_respondents.append({
+                'href': '/respondent-details/' + str(respondent['id']),
+                'name': respondent['firstName'] + ' ' + respondent['lastName'],
+                'email': respondent['emailAddress'],
+                'status': status_enum_to_string(respondent['status']),
+                'status_class': status_enum_to_class(respondent['status'])
+            })
+        except KeyError:
+            return []
```

Comment:
Looking at this again, this should probably change.  If any single one of the respondents has something wrong with it, then you get nothing back.   Is there a way we can gracefully degrade this?  Maybe with empty fields or  'Not found' values?  At the minimum a log line should happen if we're going to return nothing.

Reply:
That was the intention, but I don't really know the chances of getting data back with poor integrity.  I think on reflection I'll switch it to just not add the entry and log that it was skipped, so we still get the rest of the list.

Reply:
That's fair.  In the past there was a ton of 'we didn't get what we wanted perfectly?  500 error'.  Which makes developing pretty miserable in places.  I think graceful degradation is preferable where ever possible to not blast users out of their journey